Parsers/ASimWebSession/Parsers/vimWebSessionF5ASM.yaml

```diff
@@ -0,0 +1,261 @@
+Parser:
+  Title: Web Session ASIM parser for F5 BIG-IP Application Security Manager (ASM)
+  Version: '0.1.0'
+  LastUpdated: Oct 12 2023
+Product:
+  Name: F5 BIG-IP Application Security Manager (ASM)
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: F5 Big IP ASM Documentation
+  Link: 
+    https://techdocs.f5.com/en-us/bigip-15-0-0/external-monitoring-of-big-ip-systems-implementations/event-messages-and-attack-types.html
+    https://docplayer.net/9349757-Common-event-format-configuration-guide.html
+Description: |
+  This ASIM parser supports normalizing F5 Big IP ASM(Application Security Manager) logs to the ASIM Web Session normalized schema. F5 Big IP ASM events are captured through F5 Network data connector which ingests attack events and data into Microsoft Sentinel in CEF format over Syslog using F5's technology partner Arcsight.
+ParserName: vimWebSessionF5ASM
+EquivalentBuiltInParser: _Im_WebSession_F5ASM
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let DvcActionLookup = datatable (DeviceAction: string, DvcAction: string)
+  [
+      "Blocked", "Deny",
+      "blocked", "Deny",
+      "Passed", "Allow",
+      "passed", "Allow",
+      "Alerted", "Deny",
+      "alerted", "Deny"
+  ];
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+      [
+      "0", "Low",
+      "1", "Low",
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
+  ];
+  let parser = (
+      starttime: datetime=datetime(null),
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]),
+      ipaddr_has_any_prefix: dynamic=dynamic([]),
+      url_has_any: dynamic=dynamic([]),
+      httpuseragent_has_any: dynamic=dynamic([]),
+      eventresultdetails_in: dynamic = dynamic([]),
+      eventresult: string='*',
+      disabled: bool=false
+      ) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let DeviceEventClassIDList = dynamic(["Brute Force Attack", "IP Enforcer Attack", "Web Scraping Attack", "DoS Attack"]);
```

Comment:
We only found these values? Could there be more like these in 'DeviceEventClassID'?

Reply:
No. As only these many values are given in document.